### PR TITLE
ECS/CLIENT-GAME Add debug mode for collision rect

### PR DESCRIPTION
**Pull request by KitetsuK**
Add player shoot
MINOR

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**
Add debug mode for collision rectangle
Add player shoot

* **Other information**:

### CODE REVIEW

File: src/ECS/ECSCustomTypes.hpp [hunk 1]
```diff
@@ -34,6 +34,16 @@ namespace Types {
...
 
+    struct Velocity {
+            float speedX;
+            float speedY;
+    };
+
     struct Player { };
 
+    struct Missiles {
+            // maybe enum better ?
```
Comment (kw4ntiq) [inline src/ECS/ECSCustomTypes.hpp]: Yep

File: src/Client/Systems/Events/EventsSystems.cpp [hunk 2]
```diff
@@ -55,6 +55,62 @@ namespace Systems {
...
+        Registry::components<Types::Position> arrPosition =
+            Registry::getInstance().getComponents<Types::Position>();
+        Registry::components<Types::Rect> arrRect =
+            Registry::getInstance().getComponents<Types::Rect>();
+
+        std::vector<std::size_t> ids = arrPlayer.getExistingsId();
+
+        if (Raylib::isKeyDown(Raylib::KeyboardKey::KB_SPACE)) {
+            for (auto &id : ids) {
+                if (arrPosition.exist(id) && arrRect.exist(id)) {
```
Comment (kw4ntiq) [inline src/Client/Systems/Events/EventsSystems.cpp]: oh le if for if

File: src/Client/Systems/Events/EventsSystems.hpp [hunk 3]
```diff
@@ -15,6 +15,7 @@ namespace Systems {
     namespace EventsSystems {
         void playerMovement(std::size_t /*unused*/, std::size_t /*unused*/);
         void changeScene(std::size_t /*unused*/, std::size_t /*unused*/);
+        void shootBullet(std::size_t, std::size_t);
```
Comment (kw4ntiq) [inline src/Client/Systems/Events/EventsSystems.hpp]:
```suggestion
        void shootBullet(std::size_t /*unused*/, std::size_t /*unused*/);
```

File: src/ECS/Systems/Systems.cpp [hunk 4]
```diff
@@ -101,20 +102,68 @@ namespace Systems {
...
+    constexpr float playerHeight             = 10.0F;
+
+    constexpr int ennemyData                  = 10;
+    constexpr float ennemyWidth               = 20.F;
+    constexpr float ennemyHeight              = 20.F;
+    const Types::Dammage ennemyDammage        = {20};
+    const Types::Rect ennemySpriteRect        = {2, 67, 30, 32};
+    const Types::CollisionRect ennemyCollRect = {17, 18};
 
     void init(std::size_t managerId, std::size_t systemId)
```
Comment (kw4ntiq) [inline src/ECS/Systems/Systems.cpp]: ca serait bien de remettre les trucs de dev genre le player qui court et tout

File: src/Client/Systems/Events/EventsSystems.cpp [hunk 5]
```diff
@@ -55,6 +55,62 @@ namespace Systems {
...
+                        .getComponents<Types::Velocity>()
+                        .insertBack(velocity);
+                    Registry::getInstance()
+                        .getComponents<Types::Health>()
+                        .insertBack(health);
+                    Registry::getInstance()
+                        .getComponents<Types::Dammage>()
+                        .insertBack(dammage);
+                    Registry::getInstance().setToFrontLayers(entityId);
+                }
```
Comment (guillaumeAbel) [inline src/Client/Systems/Events/EventsSystems.cpp]: 🤮

**Review by kw4ntiq — CHANGES_REQUESTED** [review]

File: src/ECS/ECSCustomTypes.hpp (hunk 1 above)
Comment (guillaumeAbel) [inline src/ECS/ECSCustomTypes.hpp]: yes yes yes

File: src/ECS/Systems/Systems.cpp [hunk 6]
```diff
@@ -101,20 +102,68 @@ namespace Systems {
...
+        std::vector<std::size_t> ids = arrCollisionRect.getExistingsId();
+
+        for (auto &id : ids) {
+            if (arrPosition.exist(id) && !arrRectangleShape.exist(id)) {
+                Registry::getInstance()
+                    .getComponents<Types::RectangleShape>()
+                    .insert(
+                        id,
+                        {arrCollisionRect[id].width + 2,
+                         arrCollisionRect[id].height + 2});
```
Comment (guillaumeAbel) [inline src/ECS/Systems/Systems.cpp]: why don't fit exactly the collisionRect

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

File: src/Client/Systems/Events/EventsSystems.cpp (hunk 5 above)
Comment (Saverio976) [reply src/Client/Systems/Events/EventsSystems.cpp]: C'est quoi ça

**Review by guillaumeAbel — APPROVED** [review]

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]
Debug mode is launched by default

**Review by Saverio976 — APPROVED** [review]

**Review by guillaumeAbel — APPROVED** [review]

**Review by kw4ntiq — APPROVED** [review]

**Pull request merged**